[BUG] Indexing reports missing required metadata field "record_id"

## Description

All of our METS files report a missing required metadata field "record_id" although they contain a `mods:recordIdentifier` tag.
METS files from SLUB Dresden seem to raise the same error message:
```
vendor/bin/typo3 kitodo:index --pid 3 --solr dlfCore2 --doc https://digital.slub-dresden.de/data/kitodo/Abenaud_399196951-1823/Abenaud_399196951-1823_mets.xml -vvv
Cannot load Xdebug - it was already loaded

Index single document into database and Solr.
=============================================

Indexing 13 ("https://digital.slub-dresden.de/data/kitodo/Abenaud_399196951-1823/Abenaud_399196951-1823_mets.xml") on PID 3.
----------------------------------------------------------------------------------------------------------------------------

Thu, 20 Mar 2025 11:45:24 +0000 [ERROR] request="406fb3f8d96cd" component="Kitodo.Dlf.Validation.DocumentValidator": Missing required metadata field "record_id". 
Indexing 13 ("https://digital.slub-dresden.de/data/kitodo/Abenaud_399196951-1823/Abenaud_399196951-1823_mets.xml") on Solr core 1.
----------------------------------------------------------------------------------------------------------------------------------

Thu, 20 Mar 2025 11:45:24 +0000 [ERROR] request="406fb3f8d96cd" component="Kitodo.Dlf.Validation.DocumentValidator": Missing required metadata field "record_id". 

 [ERROR] ERROR: Document with UID "13" could not be indexed on Solr core 1. There are missing mandatory fields (at least
         one of those: document_format,record_id) in this document.
```

--- a/Classes/Common/Indexer.php
+++ b/Classes/Common/Indexer.php
@@ -361,7 +361,7 @@ class Indexer
                 if(strtotime($metadata['date'][0])) {
                     $solrDoc->setField('date', self::getFormattedDate($metadata['date'][0]));
                 }
-                $solrDoc->setField('record_id', $metadata['record_id'][0]);
+                $solrDoc->setField('record_id', $metadata['record_id'][0] ?? '');
                 $solrDoc->setField('purl', $metadata['purl'][0] ?? '');
                 $solrDoc->setField('location', $document->getLocation());
                 $solrDoc->setField('urn', $metadata['urn']);
@@ -703,8 +703,9 @@ class Indexer
     {
         if (is_array($authors)) {
             foreach ($authors as $i => $author) {
-                $splitName = explode(pack('C', 31), $author);
-                $authors[$i] = $splitName[0];
+                if (is_array($author)) {
+                    $authors[$i] = $author['name'];
+                }
             }
         }
         return $authors;

--- a/Classes/Validation/DocumentValidator.php
+++ b/Classes/Validation/DocumentValidator.php
@@ -80,10 +80,13 @@ class DocumentValidator
      */
     public function hasAllMandatoryMetadataFields(): bool
     {
-        foreach ($this->requiredMetadataFields as $requiredMetadataField) {
-            if (empty($this->metadata[$requiredMetadataField][0])) {
-                $this->logger->error('Missing required metadata field "' . $requiredMetadataField . '".');
-                return false;
+        //TODO: add the list of types that do not require mandatory metadata fields
+        if ($this->metadata['type'][0] !== 'chapter' && $this->metadata['type'][0] !== 'page') {
+            foreach ($this->requiredMetadataFields as $requiredMetadataField) {
+                if (empty($this->metadata[$requiredMetadataField][0])) {
+                    $this->logger->error('Missing required metadata field "' . $requiredMetadataField . '".');
+                    return false;
+                }
             }
         }
         return true;

--- a/Tests/Unit/Validation/DocumentValidatorTest.php
+++ b/Tests/Unit/Validation/DocumentValidatorTest.php
@@ -30,8 +30,25 @@ class DocumentValidatorTest extends UnitTestCase
     public function passesHasAllMandatoryMetadataFields()
     {
         $metadata = [
+            'type' => [
+                'newspaper'
+            ],
             'record_id' => [
                 'xyz'
+            ]
+        ];
+        $documentValidator = new DocumentValidator($metadata, $this->getRequiredMetadataFields());
+        self::assertTrue($documentValidator->hasAllMandatoryMetadataFields());
+    }
+
+    /**
+     * @test
+     */
+    public function passesHasNotMandatoryMetadataFieldsButType()
+    {
+        $metadata = [
+            'type' => [
+                'chapter'
             ]
         ];
         $documentValidator = new DocumentValidator($metadata, $this->getRequiredMetadataFields());
@@ -46,6 +63,9 @@ class DocumentValidatorTest extends UnitTestCase
         $metadata = [
             'document_format' => [
                 'METS'
+            ],
+            'type' => [
+                'newspaper'
             ]
         ];
         $documentValidator = new DocumentValidator($metadata, $this->getRequiredMetadataFields());

